### DISCUSSION TITLE
Split data to perform NER

### DISCUSSION THREAD

**Discussion by mandar-avhad** [explosion/spaCy#11527] [Help: Coding & Implementations] [answered]
Hi, I want to perform NER on a data.
What is the proper way to split the data?
Are there any stratified or other methods to do this?

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @mandar-avhad , in general, spaCy accepts the standard train / dev / test split during training and evaluation. This means that you have to split your data, preferably in a [serialized DocBin format](https://spacy.io/api/data-formats#training), beforehand. You can check a sample conversion script [here](https://github.com/explosion/projects/blob/v3/pipelines/ner_demo/scripts/convert.py).

If you want to do stratified splits, you can implement a custom [`Corpus`](https://spacy.io/api/corpus) that gives you the correct batch during training. You can also check out an [example project](https://github.com/explosion/projects/tree/v3/tutorials/parser_low_resource) that does cross-validation (note: it may not be the most efficient solution).

**Reply by mandar-avhad** [reply]
Yes, I want to split my NER json data into train/dev/test before converting it to spacy format.
Could you share some specific references, please?
I want to know the technique which can be used for NER

**Reply by ljvmiranda921** [reply]
Hi @mandar-avhad , have you taken a look at the sample project in the initial response already? You can split your NER JSON using Python programmatically. Load the dataset and split the resulting data structure (you can use libraries like `json` or `srsly` for that). Once you've partitioned them into train / test / split, then you can convert them into a `DocBin` object. Use this [`ner_demo` project conversion script](https://github.com/explosion/projects/blob/v3/pipelines/ner_demo/scripts/convert.py) as a starting point.
